### PR TITLE
Password & tenant settings api

### PR DESCRIPTION
**Pull request by Bars92**
## Description
add password get / update
add tenant settings get / update
updated readme
add tests

## Must
- [x] Tests
- [x] Documentation (if applicable)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -814,6 +831,36 @@ Certifcate contents
...
+reuse := false
+reuseAmount := 10
+lock := true
+lockAttempts := 5
+emailServiceProvider := "Descope"
+emailSubject := "<your-reset-email-subject>"
+emailBody := "<your-reset-email-html>"
+emailBodyPlainText := "<your-reset-email-text>"
+useEmailBodyPlainText := true
+err := descopeClient.Management.Password().ConfigureSettings(context.Background(), tenantID, enabled, minLength, lowercase, uppercase, number, nonNumber, expiration, expirationWeeks, reuse, reuseAmount, lock, lockAttempts, emailServiceProvider, emailSubject, emailBody, emailBodyPlainText, useEmailBodyPlainText)
```
Comment (Bars92) [inline README.md:861]: this should probably be a struct by now

File: descope/internal/mgmt/tenant.go [hunk 2]
```diff
@@ -105,6 +105,42 @@ func (t *tenant) SearchAll(ctx context.Context, options *descope.TenantSearchOpt
...
+		"selfProvisioningDomains":    settings.SelfProvisioningDomains,
+		"enabled":                    settings.SessionSettingsEnabled,
+		"sessionTokenExpiration":     settings.SessionTokenExpiration,
+		"refreshTokenExpiration":     settings.RefreshTokenExpiration,
+		"sessionTokenExpirationUnit": settings.SessionTokenExpirationUnit,
+		"refreshTokenExpirationUnit": settings.RefreshTokenExpirationUnit,
+		"inactivityTime":             settings.InactivityTime,
+		"inactivityTimeUnit":         settings.InactivityTimeUnit,
+		"enableInactivity":           settings.EnableInactivity,
+		"rotateJwt":                  settings.RotateJwt,
```
Comment (Bars92) [inline descope/internal/mgmt/tenant.go:137]: remove

**Comment by aviadl** [comment]
@Bars92 you have conflicts

**Comment by Bars92** [comment]
@aviadl dont forget this one

**Review by aviadl — APPROVED** [review]

File: descope/sdk/mgmt.go [hunk 3]
```diff
@@ -48,6 +48,13 @@ type Tenant interface {
 	// and results. Using nil will result in a filter-less query with a set amount of
 	// results.
 	SearchAll(ctx context.Context, options *descope.TenantSearchOptions) ([]*descope.Tenant, error)
+
+	// Get tenant settings for a tenant by id.  Tenant ID is required.
+	GetSettings(ctx context.Context, tenantID string) (*descope.TenantSettings, error)
```
[the remark this answers is not in the document]
Comment (Bars92) [reply descope/sdk/mgmt.go:53]: this is not just session currently

**Pull request merged**